lib/rules/jsx-no-target-blank.js

```diff
@@ -12,51 +12,75 @@ const linkComponentsUtil = require('../util/linkComponents');
 // Rule Definition
 // ------------------------------------------------------------------------------
 
-function isTargetBlank(attr) {
-  return attr.name
-    && attr.name.name === 'target'
-    && attr.value
-    && ((
-      attr.value.type === 'Literal'
-      && attr.value.value.toLowerCase() === '_blank'
-    ) || (
-      attr.value.type === 'JSXExpressionContainer'
-      && attr.value.expression
-      && attr.value.expression.value
-      && String(attr.value.expression.value).toLowerCase() === '_blank'
-    ));
+function lastIndexMatching(arr, condition) {
+  return arr.map(condition).lastIndexOf(true);
+}
+
+function hasTargetBlank(node, warnOnSpreadAttributes, spreadAttributeIndex) {
+  const targetIndex = lastIndexMatching(node.attributes, (attr) => attr.name && attr.name.name === 'target');
+  const foundTargetBlank = targetIndex !== -1 && ((attr) => {
+    if (!attr.value) {
+      return false;
+    }
+    const value = attr.value;
+    if (value.type === 'Literal' && typeof value.value === 'string' && value.value.toLowerCase() === '_blank') {
+      return true;
+    }
+    if (value.type === 'JSXExpressionContainer') {
+      const expr = value.expression;
+      if (expr.type === 'Literal' && typeof expr.value === 'string' && expr.value.toLowerCase() === '_blank') {
+        return true;
+      }
+      if (expr.type === 'ConditionalExpression') {
+        if (expr.alternate.type === 'Literal' && expr.alternate.value && expr.alternate.value.toLowerCase() === '_blank') {
+          return true;
+        }
+        if (expr.consequent.type === 'Literal' && expr.consequent.value && expr.consequent.value.toLowerCase() === '_blank') {
+          return true;
+        }
+      }
+    }
+    return false;
+  })(node.attributes[targetIndex]);
```

Comment:
this IIFE is a bit weird here. can it be refactored?

Reply:
Hm, yes maybe it's a bit too terse. I will break out the function, which might also make this a little faster since the rule won't have to recreate the function.